File: src/Views/FormViews/SignUpFormView.cs
using CineComplex.Classes;
using CineComplex.Classes.Base;
using CineComplex.Interfaces;
using CineComplex.ViewModels.FormViewModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CineComplex.Views.FormViews
{
    public class SignUpFormView : AViewBase<SignUpFormView>, IView
    {
        private int _choice = 0;
        public int Choice { get => _choice; set => _choice = value; }
        public List<string> MenuList { get; set; }

        public void LoadMenuList()
        {
            Instance.MenuList = new List<string>()
            {
                "1. Enter Name",
                "2. Enter Email",
                "3. Enter Contact",
                "4. Enter Password",
                "5. Sign Up",
                "6. Reset",
                "7. Exit"
            };
        }

        public void View()
        {
            Instance.LoadMenuList();

            do
            {
                Console.Clear();
                Console.WriteLine("\t----- !!! Salam Hindusthan !!! -----");
                Console.WriteLine("================================================");

                Console.WriteLine("\nSign Up Form - CineComplex");
                Console.WriteLine("-------------------------------------------------");

                SignUpFormView.Instance.ShowFormData();
                Console.WriteLine();

                Console.WriteLine("\nMenu : ");
                Console.WriteLine("---------------");

                foreach (string instr in SignUpFormView.Instance.MenuList)
                {
                    Console.WriteLine(instr);
                }

                Console.Write("Enter Your Choice : ");

                int.TryParse(Console.ReadLine(), out _choice);
                switch (Choice)
                {
                    case 1:
                        Console.Write("Enter User Name: ");
                        SignUpFormViewModel.Instance.UserName = Console.ReadLine();
                        break;
                    case 2:
                        Console.Write("Enter Email: ");
                        SignUpFormViewModel.Instance.Email = Console.ReadLine();
                        break;
                    case 3:
                        Console.Write("Enter Contact: ");
                        SignUpFormViewModel.Instance.Contact = Console.ReadLine();
                        break;
                    case 4:
                        Console.Write("Enter Password: ");
                        SignUpFormViewModel.Instance.Password = Console.ReadLine();
                        break;

                    case 5:

                        Result<bool> result = SignUpFormViewModel.Instance.CreateUserCommand();
                        Console.Write(result.Message);
                        Console.ReadKey();

                        if (result.IsSuccessful)
                        {
                            //SignUpFormViewModel.Instance.ResetFormCommand();
                        }

                        break;

                    case 6:
                        SignUpFormViewModel.Instance.ResetFormCommand();
                        break;

                    case 7:
                        SignUpFormViewModel.Instance.ResetFormCommand();
                        break;

                    default:
                        Console.WriteLine("Please enter the valid Choice .....");
                        break;
                }


            } while (Choice != Instance.MenuList.Count);
        }

        public void ShowFormData()
        {
            Console.WriteLine($"Entered Name : {SignUpFormViewModel.Instance.UserName}");
            Console.WriteLine($"Entered Email : {SignUpFormViewModel.Instance.Email}");
            Console.WriteLine($"Entered Contact : {SignUpFormViewModel.Instance.Contact}");
            Console.WriteLine($"Entered Password : {SignUpFormViewModel.Instance.Password}");
        }
    }
}

File: src/Views/FormViews/SignUpFormView.py
from src.Classes.Base.ABaseView import ABaseView
from src.ViewModels.FormViewModels.SignUpFormViewModel import SignUpFormViewModel


class SignUpFormView(ABaseView):

    def View(self):

        while self.Choice != len(self.MenuList):

            # TODO: Create definition for this part. Do not repeat yourself
            print("\n\t----- !!! Salam Hindusthan !!! -----")
            print("================================================")
            print("🏠 Sign Up - CineComplex")
            print("-------------------------------------------------")
            self.ShowFormData()
            print("-------------------------------------------------")
            print("\nMenu:")
            print("---------------")

            for instr in self.MenuList:
                print(instr)

            try:
                self.Choice = int(input("🎯 Enter Your Choice: ").strip())
            except Exception as e:
                print(f"Error : {e}")
                print("❌ Invalid choice. Please enter a number.")
                continue

            app = self.Menu.get(self.Choice)

            if app is not None:
                app()

    def __init__(self):
        super().__init__()
        self.MenuList = [
            "1. Enter Name",
            "2. Enter Email",
            "3. Enter Contact",
            "4. Enter Password",
            "5. Sign Up",
            "6. Reset Form ",
            "7. Exit",
        ]

        self.Menu = {
            1: SignUpFormViewModel().SetUserName(),
            2: SignUpFormViewModel().SetUserEmail(),
            3: SignUpFormViewModel().SetUserContact(),
            4: SignUpFormViewModel().SetUserPassword(),
            5: SignUpFormViewModel().SignUp(),
            6: SignUpFormViewModel().ResetFormData(),
        }

    @staticmethod
    def ShowFormData():
        print(f"Enter Name:{SignUpFormViewModel().Name}")
        print(f"Enter Email:{SignUpFormViewModel().Email}")
        print(f"Enter Contact:{SignUpFormViewModel().Contact}")
        print(f"Enter Password:{SignUpFormViewModel().Password}")
